File: src/main/java/recipestore/input/InputModule.java
package recipestore.input;

import com.google.inject.AbstractModule;
import com.google.inject.Provides;
import recipestore.ResourceLoader;
import recipestore.db.triplestore.FileBasedTripleStoreDAO;
import recipestore.db.triplestore.TripleStoreDAO;

import java.io.IOException;
import java.io.InputStream;
import java.nio.file.Files;
import java.nio.file.Paths;
import java.nio.file.StandardOpenOption;

import static recipestore.ResourceLoader.Resource.triplestore;

public class InputModule extends AbstractModule {

    private static final String DATASET_FILE_LOC = ResourceLoader.get.apply(triplestore,
            "input-file").orElse(null);


    @Provides
    public static String providesDatasetName() {
        return "recipe";
    }


    @Provides
    public static InputStream providesDatasetStream() {
        try {
            return Files.newInputStream(Paths.get(DATASET_FILE_LOC), StandardOpenOption.READ);
        } catch (IOException e) {
            throw new RuntimeException("Failed initializing module. Input quad file is mandatory");
        }
    }

    @Override
    protected void configure() {
        bind(TripleStoreDAO.class).to(FileBasedTripleStoreDAO.class);
    }
}

File: src/main/scala/recipestore/input/InputModule.scala
package recipestore.input

import java.io.{IOException, InputStream}
import java.nio.file.{Files, Paths, StandardOpenOption}

import com.google.inject.Provides
import com.google.inject.name.Names
import recipestore.db.triplestore.{FileBasedTripleStoreDAO, TripleStoreDAO}
import recipestore.{AppResource, CommonModule, ResourceLoader}

object InputModule {
  private val DATASET_FILE_LOC: String = ResourceLoader.apply(AppResource.TriplestoreResource, "input-file").get

  @Provides def providesDatasetStream: InputStream = {
    try {
      return Files.newInputStream(Paths.get(DATASET_FILE_LOC), StandardOpenOption.READ)
    }
    catch {
      case e: IOException => {
        throw new RuntimeException("Failed initializing module. Input quad file is mandatory")
      }
    }
  }
}

class InputModule extends CommonModule {
  override protected def configure() {
    super.configure()
    bind(classOf[TripleStoreDAO]).to(classOf[FileBasedTripleStoreDAO])
    bind(classOf[String]).annotatedWith(Names.named("datasetName")).toInstance("recipe")
  }
}